x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedNodeSelector.java

```diff
@@ -120,37 +121,34 @@ private AssignmentFailure checkAssignment() {
 
     @Nullable
     private AssignmentFailure verifyIndicesActive() {
-        for (String index : datafeedIndices) {
```

Comment:
The other option is to do this foreach loop but simply continue if `allow_no_indices` is true. 

@dimitris-athanasiou @droberts195 what do you think? I am not sure if we want to do this with `allow_no_indices` being true as then we might have an issue later on when we attempt to extract fields. 

This way enforces at least one concrete index to exist (for now)

Reply:
I think it makes sense to require at least one concrete index for now.